### PR TITLE
Add pretty reporter and --format option

### PR DESCRIPTION
**Pull request by scalvert**
Adds new `--format` option, and provides two formatters
- `markdown` - default formatter that writes to a md file
- `pretty` - outputs to the console

### CODE REVIEW

File: __tests__/cli-test.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import '@microsoft/jest-sarif';
+import { readFileSync } from 'fs';
```
Comment (rwjblue) [inline __tests__/cli-test.ts]:
```suggestion
```

File: __tests__/cli-test.ts [hunk 2]
```diff
@@ -37,11 +40,29 @@ describe('cli-test', () => {
...
     expect(result.exitCode).toEqual(0);
-    expect(JSON.parse(result.stdout)).toBeValidSarifLog();
+    expect(result.stdout).toMatch(
+      /Embroider preflight check written to .*\/embroider-preflight.md/
+    );
+  });
+
+  it('can invoke the preflight command using the pretty formatter', async () => {
+    const result = await run(['preflight', '--format', 'pretty']);
+    debugger;
```
Comment (rwjblue) [inline __tests__/cli-test.ts]:
```suggestion
```

Surprised this didn't fail linting 🤔

File: package.json [hunk 3]
```diff
@@ -21,7 +21,7 @@
     "copy-files": "cp -R ./src/templates ./lib && cp -R src/codemods ./lib",
     "lint": "eslint . --cache --ext .ts",
     "prepare": "yarn build",
-    "test": "jest --runInBand"
+    "test": "jest --runInBand --silent"
```
Comment (rwjblue) [inline package.json]: What does `--silent` do for us?

File: __tests__/cli-test.ts (hunk 2 above)
Comment (scalvert) [reply __tests__/cli-test.ts]: Gah. Indeed. I'll take a look at linting setup.

File: package.json (hunk 3 above)
Comment (scalvert) [reply package.json]: It's _supposed_ to suppress the console output in tests, but it doesn't appear to do so (perhaps I misinterpreted). I can remove.

**Review by rwjblue — APPROVED** [review]

**Pull request merged**